### PR TITLE
feat: dynamic deposit

### PR DESCRIPTION
**Pull request by Ad96el**
## fixes [2636](https://github.com/KILTprotocol/ticket/issues/2636)

Documentation for the [PR](https://github.com/KILTprotocol/kilt-node/pull/507).

### CODE REVIEW

File: docs/develop/02_chain/02_pallets/01_did.md [hunk 1]
```diff
@@ -99,8 +99,9 @@ All of them have to be authenticated using the DID that is updated following the
 
 ## What About the Deposit?
 
-When writing a DID to the chain the submitter of the extrinsic has to pay a deposit, currently 2 KILT.
-This is to incentivize deleting unused DIDs to save storage on the chain.
+When writing a DID to the chain the submitter of the extrinsic has to pay a deposit. The base deposit are currently 2 KILT.
```
Comment (ntn-x2) [inline docs/develop/02_chain/02_pallets/01_did.md]:
```suggestion
When writing a DID to the chain the submitter of the extrinsic has to pay a deposit.
The base deposit is currently 2 KILT.
```

File: docs/develop/02_chain/02_pallets/01_did.md [hunk 2]
```diff
@@ -99,8 +99,9 @@ All of them have to be authenticated using the DID that is updated following the
...
+For additional used storage, for example by adding more service endpoints, the deposit increases.
```
Comment (ntn-x2) [inline docs/develop/02_chain/02_pallets/01_did.md]:
```suggestion
For additional used storage, for example by adding more services, more tokens are taken as deposit, depending on the amount of additional storage taken.
```

File: docs/develop/02_chain/02_pallets/01_did.md [hunk 3]
```diff
@@ -99,8 +99,9 @@ All of them have to be authenticated using the DID that is updated following the
...
+Releasing memory reduces the deposit. This is to incentivize deleting unused DIDs to save storage on the chain.
```
Comment (ntn-x2) [inline docs/develop/02_chain/02_pallets/01_did.md]:
```suggestion
Freeing up memory reduces the deposit.
This is to incentivize deleting unused DIDs or keys to reduce the total storage of the chain.
```

**Review by ntn-x2 — COMMENTED** [review]
One-line-one-sentence 😀

File: docs/develop/02_chain/02_pallets/01_did.md [hunk 4]
```diff
@@ -99,8 +99,11 @@ All of them have to be authenticated using the DID that is updated following the
 
 ## What About the Deposit?
 
-When writing a DID to the chain the submitter of the extrinsic has to pay a deposit, currently 2 KILT.
-This is to incentivize deleting unused DIDs to save storage on the chain.
+When writing a DID to the chain the submitter of the extrinsic has to pay a deposit.
+The base deposit is currently 2 KILT.
```
Comment (weichweich) [inline docs/develop/02_chain/02_pallets/01_did.md:103]: This suggests that it's 2 KILT exactly. Either give an exact number or make clear that it's an approximation.

File: docs/develop/02_chain/02_pallets/01_did.md [hunk 5]
```diff
@@ -99,8 +99,11 @@ All of them have to be authenticated using the DID that is updated following the
...
+For additional used storage, for example by adding more services, more tokens are taken as deposit, depending on the amount of additional storage taken.
+Freeing up memory reduces the deposit.
```
Comment (weichweich) [inline docs/develop/02_chain/02_pallets/01_did.md]: Memory is working memory, RAM while storage is persistent storage. Let's keep one wording here.

```suggestion
Freeing up storage reduces the deposit.
```

**Review by weichweich — CHANGES_REQUESTED** [review]
minor requests

File: docs/develop/02_chain/02_pallets/01_did.md (hunk 4 above)
Comment (Ad96el) [reply docs/develop/02_chain/02_pallets/01_did.md:103]: The base deposit is exactly [2 KILT](https://github.com/KILTprotocol/kilt-node/blob/develop/runtimes/common/src/constants.rs#L347) :thinking:

Comment (weichweich) [reply docs/develop/02_chain/02_pallets/01_did.md:103]: Ah Sorry... it wasn't exactly 2 KILT before. Good to know. 😁

**Review by weichweich — APPROVED** [review]

**Pull request merged**